include/boost/geometry/algorithms/detail/is_valid/linear.hpp

```diff
@@ -165,15 +164,12 @@ class is_valid
             return visitor.template apply<no_failure>();
         }
 
-        typedef per_linestring<VisitPolicy, Strategy> per_ls;
+        using per_ls = per_linestring<VisitPolicy, Strategy>;
 
-        return detail::check_iterator_range
-            <
-                per_ls,
-                false // do not check for empty multilinestring (done above)
-            >::apply(boost::begin(multilinestring),
-                     boost::end(multilinestring),
-                     per_ls(visitor, strategy));
+        return boost::end(multilinestring) == std::find_if(
+            boost::begin(multilinestring),
+            boost::end(multilinestring),
+            per_ls(visitor, strategy));
```

Comment:
The behavior is changed here. `false` was returned for empty range. But in this case this is probably irrelevant, right?

Reply:
Yes, here is irrelevant.